data/favorites/src/main/java/de/mm20/launcher2/favorites/FavoritesRepository.kt

```diff
@@ -286,6 +291,18 @@ internal class FavoritesRepositoryImpl(
         }
     }
 
+    override fun getRanksByLaunchCount(limit: Int): Flow<List<Pair<Int, SavableSearchable>>> {
```

Comment:
It's unnecessary to query the entire `SavableSearchable` object. Query the keys only to reduce the overhead.
Also don't use `Pair` in public APIs, it makes it difficult to understand, what exactly is returned here. Use a new, more descriptive data class as return type.

Reply:
Nevermind, it's used. Let's see if we can optimize that away.

Reply:
No you are right, it should be enough to just return something like a new dataclass just containing `launchCount` and `key`. After all, reordering just happens by keys, that are ordered by their launch count

Reply:
But we could use the `key`, `type`, and `launchCount` fields from the `SavedSearchableEntity` and put those in a new data class, so we don't need to deserialize the actual `Searchable`

Reply:
This new class wouldn't be an entity though, right? Does it still make sense to put it into `database.entities`?